include/PathResolver.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2017-2018  Waterloo Quantitative Consulting Group, Inc. 
+// See COPYING and LICENSE files at project root for more details. 
+
+/**
+ * @file PathResolver.hpp
+ * @brief Defines the PathResolver class that allows for the resolution of paths
+ * relative to the project root path.
+ */
+
+#pragma once
+#include <string>
+#include <vector>
+#include <memory>
+#include <iostream>
+#include <whereami.h>
+#include <boost/algorithm/string.hpp>
+
+using boost::algorithm::find_all;
+using boost::algorithm::join;
+using boost::algorithm::replace_last;
+using boost::algorithm::replace_first;
+using boost::algorithm::trim_right;
+using boost::iterator_range;
+using std::string;
+using std::vector;
+using std::cout;
+using std::endl;
+using std::numeric_limits;
+using std::abs;
+using std::shared_ptr;
+using std::unique_ptr;
+
+namespace blitzdg {
+  class PathResolver {
+  
+    string PathDelimeter = "/";
+    shared_ptr<string> ExePath = nullptr;
+    unique_ptr<vector<string>> InputPathVec = nullptr;
+
```

Comment:
You are using inline initialization here. Typically these default values are set in the constructor. Is it necessary to use a shared_ptr to string and a unique_ptr to vector<string>? The types string and vector manage their own memory, so why not just use composition a store a string and a vector<string>?

Reply:
Good point, fixed.